src/aoefit.jl

```diff
@@ -327,3 +327,183 @@ function fit_single_aoe_compton(h::Histogram, ps::NamedTuple; uncertainty::Bool=
+    μB = ustrip(result_corrections.μ_compton.par[2])
+    σA = ustrip(result_corrections.σ_compton.par[1])
+    σB = ustrip(result_corrections.σ_compton.par[2])
+    
+    # create pseudo priors
+     pseudo_prior = NamedTupleDist(
+        μA = Normal(mvalue(μA), muncert(μA)),
+        μB = Normal(mvalue(μB), muncert(μB)),
+        σA = Normal(mvalue(σA), muncert(σA)),
+        σB = Normal(mvalue(σB), muncert(σB)),
+    )
+    
+    # transform back to frequency space
+    f_trafo = BAT.DistributionTransform(Normal, pseudo_prior)
+    
+    # start values for MLE
+    v_init = mean(pseudo_prior)
+
+    # create loglikehood function
+    f_loglike = pars -> begin
+        
+        neg_log_likelihoods = zeros(typeof(pars.μA), length(compton_bands))
+
+        # iterate throuh all peaks (multithreaded)
+        Threads.@threads for i in eachindex(compton_bands)
+
+            # get histogram and peakstats
+            h  = peakhists[i]
+            ps = peakstats[i]
+            e = ustrip(compton_bands[i])
+            μ = f_aoe_mu(e, (pars.μA, pars.μB))
+            σ = f_aoe_sigma(e, (pars.σA, pars.σB))
+
+            # fit peak
+            try
+                A, B = fit_single_aoe_compton_with_fixed_μ_and_σ(h, μ, σ, ps; uncertainty=uncertainty)
+                neg_log_likelihoods[i] = A
+            catch e
+                @warn "Error fitting band $(compton_bands[i]): $e"
+                continue
+            end
+        end
+        return sum(neg_log_likelihoods)
+    end
+    
+    # MLE
+    opt_r = optimize(f_loglike ∘ inverse(f_trafo), f_trafo(v_init), NelderMead(), Optim.Options(time_limit = 60, show_trace=false, iterations = 1000))
+
+    converged = Optim.converged(opt_r)
+    !converged && @warn "Fit did not converge"
+
+    v_ml = inverse(f_trafo)(Optim.minimizer(opt_r))
+
+    if uncertainty && converged
+
+        f_loglike_array = p -> f_loglike((μA = p[1], μB = p[2], σA = p[3], σB = p[4]))
+
+        # Calculate the Hessian matrix using ForwardDiff
+        H = ForwardDiff.hessian(f_loglike_array, tuple_to_array(v_ml))
+
+        param_covariance = nothing
+        if !all(isfinite.(H))
+            @warn "Hessian matrix is not finite"
+            param_covariance = zeros(length(v_ml), length(v_ml))
+        else
+            # Calculate the parameter covariance matrix
+            param_covariance = inv(H)
+        end
+        if ~isposdef(param_covariance)
+            param_covariance = nearestSPD(param_covariance)
+        end
+        # Extract the parameter uncertainties
+        v_ml_err = array_to_tuple(sqrt.(abs.(diag(param_covariance))), v_ml)
+
+        # TODO: correctly combine p-values and residuals of the individual fits to the combined fit result
+        # get p-value 
+        # pval, chi2, dof = p_value(f_aoe_compton, h, v_ml)
+        # calculate normalized residuals
+        # residuals, residuals_norm, _, bin_centers = get_residuals(f_aoe_compton, h, v_ml)
```

Comment:
`p-values, chi2, dof, residuals` still need to be implemented, probably by being passed from the individual fits and combined in the combined fit

Reply:
--> a39fa4d3eae3891f81919ce4ab0d1ad54df13009